test/e2e/network/service.go

```diff
@@ -4059,12 +4109,23 @@ var _ = common.SIGDescribe("Services", func() {
 		checkOneNodePort(thirdNodeIP, true, v1.ServiceExternalTrafficPolicyCluster, deadline)
 
 		ginkgo.By("changing ExternalTrafficPolicy back to Local")
-		_, err = jig.UpdateService(ctx, func(svc *v1.Service) {
-			svc.Spec.ExternalTrafficPolicy = v1.ServiceExternalTrafficPolicyLocal
-			// Request the same healthCheckNodePort as before, to test the user-requested allocation path
-			// FIXME: we need to use a reserved HCNP here.
-			svc.Spec.HealthCheckNodePort = oldHealthCheckNodePort
+
+		// retry a few times in case some other service has taken up the static HCNP
+		// despite being reserved and might release soon
+		err = retry.OnError(retry.DefaultRetry, func(err error) bool {
+			if err != nil && strings.Contains(err.Error(), errAllocated.Error()) {
```

Comment:
when this will retry?
`UpdateService` does not seem to be able to return `errAllocated` error.

Independently, at this point in the test I don't think we want to retry, since we are willing to rely on the nodeport allocator to avoid conflicts ... 

I think this part should be the same